Detectors/CTF/workflow/src/CTFWriterSpec.cxx

```diff
@@ -616,14 +621,14 @@ void CTFWriterSpec::closeTFTreeAndFile()
           metaFileOut << ctfMetaData;
           metaFileOut.close();
           if (!TMPFileEnding.empty()) {
-            std::filesystem::rename(o2::utils::Str::concat_string(mCurrentCTFFileNameFull, TMPFileEnding), mCurrentCTFFileNameFull);
+            std::filesystem::rename(actualFileName, mCurrentCTFFileNameFull);
```

Comment:
Ah ok, I see. So now if the CTF writer would crash in the worst case after the renaming in this line then we would at least at that point have the temporary meta file and then we could more easily rename that meta file if we would want to (and if this would actually ever happen of course).
I thought first that the logic to write the meta file first and then the data file could never work, but the way you have it now is the best we can do I think. Thanks @shahor02 !